esp/tools/soapplus/httpproxy.cpp

```diff
@@ -674,25 +672,28 @@ class CSocksProxyThread : public Thread
                 return -1;
             }
 
-            len += lenread;
-            m_client->read(inbuf + len, 0, 1, lenread);
+            char inbuf2[16];
+            m_client->read(inbuf2, 0, 16, lenread);
```

Comment:
Is min size set to 0 to prevent throwing an exception on close ?

Reply:
Yes, 0 would avoid the exception, and since read will block for data, so 0 or 1 won't make much difference here.